bob-common/src/mapper.rs

```diff
@@ -121,6 +121,44 @@ impl Virtual {
         self.vdisks.get(&id).expect("vdisk not found").nodes()
     }
 
+    fn find_index(&self, offset: usize) -> Offset {
+        let mut avail = 0;
+        for i in 0..self.nodes.len() {
+            let node = &self.nodes[i];
+            if node.connection_available() {
+                if avail == offset {
+                    return Offset::Index(i);
+                }
+                avail += 1;
+            }
+        }
+        return Offset::Avail(avail);
+    }
+
+    fn find_actual_offset(&self, offset: usize) -> usize {
+        let mut res = self.find_index(offset % self.nodes.len());
+        if let Offset::Avail(avail) = res {
+            if avail > 0 {
+                let mut curr_offset = offset % avail;
+                while let Offset::Avail(avail) = res {
+                    if avail == 0 {
```

Comment:
You can move this condition to the line after `res = self.find_index(curr_offset);`. This should simplify the code

Reply:
Or you can add a function that checks that res is `Avail` and inner value is greater than zero